source/backend/api/Services/DocumentFileService.cs

```diff
@@ -89,333 +91,509 @@ public IList<T> GetFileDocuments<T>(FileType fileType, long fileId)
             {
-                // Create the pims document acquisition file relationship
-                PimsAcquisitionFileDocument newAcquisitionDocument = new PimsAcquisitionFileDocument()
-                {
-                    AcquisitionFileId = acquisitionFileId,
-                    DocumentId = uploadResult.Document.Id,
-                };
-                newAcquisitionDocument = acquisitionFileDocumentRepository.AddAcquisition(newAcquisitionDocument);
-                acquisitionFileDocumentRepository.CommitTransaction();
-
-                relationshipResponse.DocumentRelationship = mapper.Map<DocumentRelationshipModel>(newAcquisitionDocument);
-            }
+                DocumentId = pimsDocument.DocumentId,
+                ResearchFileDocumentId = newFileDocument.ResearchFileDocumentId,
+                Document = await uploadRequest.File.GetBytes(),
+                FileName = uploadRequest.File.FileName,
+                DocumentMetadata = JsonSerializer.Serialize(uploadRequest.DocumentMetadata),
+            };
+            _documentQueueRepository.Add(queueDocument);
+            _documentQueueRepository.SaveChanges();
 
-            return relationshipResponse;
+            // All good here.
+            transaction.Commit();
+
+            return;
         }
 
-        public async Task<DocumentUploadRelationshipResponse> UploadProjectDocumentAsync(long projectId, DocumentUploadRequest uploadRequest)
+        public async Task UploadProjectDocument(long projectId, DocumentUploadRequest uploadRequest)
         {
             Logger.LogInformation("Uploading document for single Project");
             User.ThrowIfNotAllAuthorized(Permissions.DocumentAdd, Permissions.ProjectEdit);
+            uploadRequest.ThrowInvalidFileSize();
 
-            // Do not call Mayan if uploaded file is empty (zero-size)
-            ValidateZeroLengthFile(uploadRequest);
+            using var transaction = _documentQueueRepository.BeginTransaction();
 
-            DocumentUploadResponse uploadResult = await documentService.UploadDocumentAsync(uploadRequest);
+            // Step 1 - Save the pimsDocument
+            PimsDocument pimsDocument = CreatePimsDocument(uploadRequest);
+            _documentQueueRepository.SaveChanges();
 
-            DocumentUploadRelationshipResponse relationshipResponse = new()
+            // Step 2 - Create the File Document
+            PimsProjectDocument newFileDocument = new()
             {
-                UploadResponse = uploadResult,
+                ProjectId = projectId,
+                DocumentId = pimsDocument.DocumentId,
             };
+            _projectRepository.AddProjectDocument(newFileDocument);
+            _documentQueueRepository.SaveChanges();
 
-            // Throw an error if Mayan returns a null document. This means it wasn't able to store it.
-            ValidateDocumentUploadResponse(uploadResult);
+            throw new NotImplementedException();
```

Comment:
@eddherrera what is going on here? why can't we save project documents?

Reply:
This was related to a column missing in the Queue Table but that has been fixed now. All good now.